waku/waku_store_v3/common.nim

```diff
@@ -0,0 +1,107 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/[options],
+  stew/results,
+  stew/byteutils,
+  nimcrypto/sha2
+import
+  ../waku_core,
+  ../common/paging
+
+const
+  WakuStoreV3Codec* = "/vac/waku/store/3.0.0"
+
+  DefaultPageSize*: uint64 = 20
+
+  MaxPageSize*: uint64 = 100
+
+type WakuStoreV3Result*[T] = Result[T, string]
+
+## Waku message digest
+
+type MessageDigest* = MDigest[256]
+
+proc computeDigest*(msg: WakuMessage): MessageDigest =
+  var ctx: sha256
+  ctx.init()
+  defer: ctx.clear()
+
+  ctx.update(msg.contentTopic.toBytes())
+  ctx.update(msg.payload)
+
+  # Computes the hash
+  return ctx.finish()
+
+## Public API types
+
+type
+  StoreRequest* = object
+    requestId*: string
+    returnValues*: bool
+
+    pubsubTopic*: Option[PubsubTopic]
+    contentTopics*: seq[ContentTopic]
+    startTime*: Option[Timestamp]
+    endTime*: Option[Timestamp]
+
+    messageHashes*: seq[WakuMessageHash]
+
+    paginationCursor*: Option[WakuMessageHash]
+    paginationForward*: PagingDirection
+    paginationLimit*: Option[uint64]
+
+  WakuMessageKeyValue* = object
+    messageHash*: WakuMessageHash
+    message*: WakuMessage
+
+  StoreResponse* = object
+    requestId*: string
+
+    statusCode*: Option[uint32]
```

Comment:
Don't think this should be an `Option` (it's a mandatory field and should result in decoding error if not present in received response)